testing/test_doctest.py

```diff
@@ -287,13 +287,50 @@ def test(self):
             )
         )
         result = testdir.runpytest("--doctest-modules")
+        result.stdout.fnmatch_lines(
+            ["*hello*", "006*>>> 1/0*", "*UNEXPECTED*ZeroDivision*", "*1 failed*"]
+        )
+
+    def test_doctest_linedata_on_property(self, testdir):
+        testdir.tmpdir.join("hello.py").write(
+            textwrap.dedent(
+                """\
+                class Sample(object):
+                    @property
+                    def some_property(self):
+                        '''
+                        >>> Sample().some_property
+                        'another thing'
+                        '''
+                        return 'something'
+                """
+            )
+        )
```

Comment:
This could be better written using `testdir.makepyfile()`, also below.
Not a big issue, but I'd be happy to push a fixup, if you agree.

Reply:
I don't mind. However, I will probably need to squash all commits after that.

Reply:
> I don't mind. However, I will probably need to squash all commits after that.

Ok, will fix this then and force-push here.